internal/manager/setup.go

```diff
@@ -474,3 +476,37 @@ func setupLicenseGetter(
 
 	return nil, nil
 }
+
+// setupKonnectConfigSynchronizer sets up Konnect config sychronizer and adds it to the manager runnables.
+func setupKonnectConfigSynchronizer(
+	ctx context.Context,
+	mgr manager.Manager,
+	configUploadPeriod time.Duration,
+	kongConfig sendconfig.Config,
+	clientsProvider clients.AdminAPIClientsProvider,
+	updateStrategyResolver sendconfig.UpdateStrategyResolver,
+	configStatusNotifier clients.ConfigStatusNotifier,
+) (*konnect.ConfigSynchronizer, error) {
+	logger := ctrl.LoggerFrom(ctx).WithName("konnect-config-synchronizer")
+	if configUploadPeriod < konnect.MinConfigUploadPeriod {
+		logger.Info("Cannot set config upload period to be smaller than the minimum upload period; use the minimum upload period instead",
+			"upload_period", configUploadPeriod, "minimum_upload_period", konnect.MinConfigUploadPeriod,
+		)
+		configUploadPeriod = konnect.MinConfigUploadPeriod
```

Comment:
How about moving this validation to `manager.Config.validateKonnect` so that if the lower period is specified, it is rejected? I think it's better to explicitly error out at startup instead of clamping to the minimal period with the info log as this may be overlooked.

Reply:
Done.